src/connect-aws/index.ts

```diff
@@ -0,0 +1,100 @@
+import { Result, URI } from "@adviser/cement";
+import { bs, getStore, Logger, NotFoundError, SuperThis, ensureSuperLog } from "@fireproof/core";
+import fetch from "cross-fetch";
+// import { Base64 } from "js-base64";
+
+export class AWSGateway implements bs.Gateway {
+  readonly sthis: SuperThis;
+  readonly logger: Logger;
+
+  constructor(sthis: SuperThis) {
+    this.sthis = ensureSuperLog(sthis, "AWSGateway");
+    this.logger = this.sthis.logger;
+  }
+
+  buildUrl(baseUrl: URI, key: string): Promise<Result<URI>> {
+    const url = baseUrl.build();
+    url.setParam("key", key);
+    return Promise.resolve(Result.Ok(url.URI()));
+  }
+
+  async destroy(): Promise<Result<void>> {
+    // Implement the destroy logic for AWS
+    return Result.Ok(undefined);
+  }
+
+  async start(url: URI): Promise<Result<URI>> {
+    await this.sthis.start();
+    this.logger.Debug().Str("url", url.toString()).Msg("start");
+    const ret = url.build().defParam("version", "v0.1-AWS").URI();
+    return Result.Ok(ret);
+  }
+
+  async close(): Promise<bs.VoidResult> {
+    return Result.Ok(undefined);
+  }
+
+  async put(url: URI, body: Uint8Array): Promise<bs.VoidResult> {
+    const { store } = getStore(url, this.sthis, (...args) => args.join("/"));
```

Comment:
this will need to go to the signed upload path for data and the websocket path for meta

Reply:
Using same put function for data and meta uploads might cause problems because they have different params inside server

Reply:
Also, this gateway takes the same implementation of buildURL from netlify for now ( looks like a placeholder for now). Earlier, these urls were constructed by executing the SAM template. What sort of mechanisms are we implementing now?

Reply:
this should be backwards compatible with the same server as the existing metaUpload / metaDownload code. to make that work I had to move the parent tracking to the store which is where it belongs anyway. I think this will simplify things overall when we get there. https://github.com/fireproof-storage/fireproof/pull/202

its ok if the impl of put switches between internal putData and putMeta functions. The good part is we don't have to worry about protocol or encoding here, just transport / storage.